cdap-app-templates/cdap-program-report/src/main/java/co/cask/cdap/report/ReportGenerationSpark.java

```diff
@@ -90,6 +95,10 @@
   private static final ExecutorService REPORT_EXECUTOR =
     new ThreadPoolExecutor(0, 3, 60L, TimeUnit.SECONDS,
                            new SynchronousQueue<>(), Threads.createDaemonThreadFactory("report-generation-%d"));
+  // User name authenticated and passed down by CDAP-Router using this key in header
+  private static final String USER_ID = "CDAP-UserId";
+  private static final String DEFAULT_USER_ID = "system";
```

Comment:
Does `system` has any special meaning? What if there is a actual user call `system`?

Reply:
no, its something i have set as a default which is used on non authenticated clusters. on authenticated clusters, we will use user name and wouldn't have to use "system" user. we can document for the case where a user wants to move from non authenticated cluster to authenticated cluster, they can rename the `system` dir to their `admin user` to continue accessing old reports for admin user.

Reply:
maybe this should be renamed as `cdap` user ?